src/main/java/org/example/Main.java

```diff
@@ -121,339 +177,256 @@ public boolean isHappy() {
+            if (Modes.EXIT.equals(input.getMode())) {
+                break;
+            }
+            if (Modes.INVALID.equals(input.getMode())) {
+                continue;
+            }
+            if (Modes.ONE_NUMBER.equals(input.getMode())) {
+                MyNumber num = new MyNumber(input.getStartNumber());
+                num.printInfo();
+                continue;
+            }
 
-                // let's make it a class constant field: private static final PROPERTIES and store them in UPPER case to avoid doing toLowerCase in a loop below
-                // also better to make it a Set<String> and check .contains instead of looping for each user-defined property
-                String[] properties = {"buzz", "duck", "palindromic",
-                        "gapful", "spy", "even", "odd", "sunny", "square", "jumping",
-                        "happy", "sad"};
-                //                 ArrayList<String> incorrect = new ArrayList<String>(); // when you create a collection it's better to use the most generic interface as a type, e.g. List<String> or even Collection<String>
-                ArrayList<String> incorrect = new ArrayList<String>();
-                for (String user: users) {
-                    boolean correct = false;
-                    for (String p : properties) {
-                        if (user.toLowerCase().equals(p)) {
-                            correct = true;
-                            break;
-                        }
-                        if (user.toLowerCase().equals("-" + p)) {
-                            correct = true;
-                            break;
-                        }
-                    }
-                    if (!correct) {
-                        incorrect.add(user);
-                    }
-                }
+            if (hasInvalidProperty(input.getProperties())) {
+                continue;
+            }
+
+            if (hasExclusiveProperty(input.getProperties())) {
+                continue;
+            }
+
+            processManyArguments(input);
+        }
+    }
+
+    private static void processManyArguments(InputStructure input) {
+        var properties = input.getProperties();
+        var startNumber = input.getStartNumber();
+        var amountOfNumbers = input.getAmountOfNumbers();
 
-                if (incorrect.size() > 0) {
-                    if (incorrect.size() == 1) {
-                        System.out.println("The property [" + incorrect.get(0).toUpperCase() + "] is wrong.");
+        System.out.println();
+        long curNum = startNumber;
+        while (amountOfNumbers > 0) {
+            MyNumber num = new MyNumber(curNum++);
+            if (properties.size() > 0) {
+                boolean needSkip = false;
+                for (var property : properties) {
+                    if (property.startsWith("-")) {
+                        needSkip |= num.getParams().contains(MyNumber.Params.valueOf(property.substring(1)));
```

Comment:
I think if we got true for one of the properties - we can break?
I've rewritten a bit, let's discuss:
```
                while (amountOfNumbers > 0) {
            MyNumber num = new MyNumber(curNum++);
            boolean print = true;
            if (properties.size() > 0) {
                for (var property : properties) {
                    print = property.startsWith("-") 
                            ? !num.containsParam(MyNumber.Params.valueOf(property.substring(1)))
                            : num.containsParam(MyNumber.Params.valueOf(property));
                    if (!print) {
                        break;
                    }
                }
            }
            if (print) {
                num.printShortInfo();
                amountOfNumbers--;
            }
        }
```

Reply:
I like using `print `variable.
This code is still quite difficult for me:
```
 property.startsWith("-") 
                            ? !num.containsParam(MyNumber.Params.valueOf(property.substring(1)))
                            : num.containsParam(MyNumber.Params.valueOf(property));
```